reconcile/statuspage/page.py

```diff
@@ -103,12 +105,24 @@ class StatusMaintenance(BaseModel):
     message: str
     schedule_start: str
     schedule_end: str
+    components: list[StatusComponent]
 
     @classmethod
-    def init_from_maintenance(cls, maintenance: MaintenanceV1) -> Self:
+    def init_from_maintenance(
+        cls,
+        maintenance: MaintenanceV1,
+        page_components: list[StatusPageComponentV1],
+    ) -> Self:
+        affected_services = [a.name for a in maintenance.affected_services]
+        affected_components = [
+            StatusComponent.init_from_page_component(c, name_override=c.display_name)
+            for c in page_components
+            if c.app.name in affected_services
+        ]
```

Comment:
this is the important part of this PR, where we find the affected components based on the affected services.